### ISSUE TITLE
node-gyp not rebuilding correctly

### ISSUE THREAD

**Issue by adelphes** [electron/electron-rebuild#404]
Native dependency detection appears to be breaking in v2.0.2.

electron-rebuild no longer correctly identifies all the native dependencies needed to be rebuilt.
Reverting back to v2.0.1 fixes the issue.

I will do some more digging into the problem, but the issue is probably related to the changes in 9d166a83bd557084993832c8270f16c688f6ee45

**Comment by malept** [comment]
It would be great if you could provide a minimal testcase for this that we could add to the testsuite, to avoid any regressions in the future.

**Comment by expressw** [comment]
I have the same issue when rebuilding native module 'serialport'.
Look at this:
https://stackoverflow.com/questions/63794519/serialport-for-electron-node-module-version-error-and-rebuild-does-not-fix#

**Comment by MarshallOfSound** [comment]
Can confirm @malept 2.0.2 is not working correctly.  Had to downgrade to 2.0.1

**Comment by malept** [comment]
Proposed fix in https://github.com/electron/electron-rebuild/pull/409, please test.

For those using yarn, use [dependency resolutions](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/) to point to the branch name.

**Comment by todbot** [comment]
As a small datapoint, PR #409 works for `node-hid` and Electron v10.1.1 https://github.com/todbot/electron-hid-test/blob/master/package.json

**Comment by adelphes** [comment]
I can confirm that #409 fixes our build. 🎉

**Comment by Githubber2021** [comment]
For some reason I can't confirm the fix. Here is a reproducible build with the latest version to compile nodegit:

https://github.com/Githubber2021/node_module_version-issue

```
git clone https://github.com/Githubber2021/node_module_version-issue
cd node_module_version-issue
npm i
npm run rebuild
...Error
```

**Comment by malept** [comment]
Not sure what you're trying to show in your repro instructions, that makes sense since you're trying to use s native module compiled for Electron 10 with AVA running with whatever Node.js version, which will never have the same  ABI version number.

**Comment by Githubber2021** [comment]
Thanks! Just a few minutes ago I noticed that Electron has a different ABI version than node.js and I don't understand why. If Electron uses nodejs 14.15.0 it has a different ABI version than the actual nodejs 14.15.0. Is that correct? I just opened a question for this on SO. Sorry if this doesn't fit here, but I have to give it a shot since it's really difficult to grasp https://stackoverflow.com/questions/64815484/why-is-there-a-mismatch-of-module-versions-between-electron-and-node-js
